File: backend/database/src/main/kotlin/de/trzpiot/example/database/domain/Post.kt
package de.trzpiot.example.database.domain

import org.neo4j.ogm.annotation.NodeEntity
import org.neo4j.ogm.annotation.typeconversion.DateLong
import java.util.*

@NodeEntity
internal data class Post(
        var id: Long? = null,
        var message: String,

        @DateLong
        var creationDate: Date
)

File: lib/domain/post.dart
class Post {
  final int id;
  final String message;
  final DateTime creationDate;

  Post._(this.id, this.message, this.creationDate);

  static Post fromQuery(dynamic data) {
    var id = data['post']['id'];
    var message = data['post']['message'];
    var creationDate = DateTime.parse(data['post']['creationDate']);
    return Post._(id, message, creationDate);
  }
}
